frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/diskDialog.ts

```diff
@@ -0,0 +1,83 @@
+import { click } from '@console/shared/src/test-utils/utils';
+import { fillInput, selectOptionByText, getSelectedOptionText } from '../utils/utils';
+import * as view from '../../views/dialogs/diskDialog.view';
+import { applyButton, saveButton } from '../../views/kubevirtDetailView.view';
+import { StorageResource } from '../utils/types';
+import { DISK_SOURCE } from '../utils/consts';
+import { waitForNoLoaders } from '../../views/wizard.view';
+
+export class DiskDialog {
+  async fillName(name: string) {
+    await fillInput(view.diskName, name);
+  }
+
+  async fillSize(size: string) {
+    if (size !== undefined && (await view.diskSize.isPresent())) {
+      await fillInput(view.diskSize, size);
+    }
+  }
+
+  async selectInterface(diskInterface: string) {
+    await selectOptionByText(view.diskInterface, diskInterface);
+  }
+
+  async selectStorageClass(storageClass: string) {
+    if (await view.diskStorageClass.isPresent()) {
+      await selectOptionByText(view.diskStorageClass, storageClass);
+    }
+  }
+
+  async getDiskSource() {
+    return getSelectedOptionText(view.diskSource);
+  }
+
+  static async selectSourceAttachDisk(PVCName: string) {
+    await selectOptionByText(view.diskSource, DISK_SOURCE.AttachDisk);
+    await selectOptionByText(view.diskPVC, PVCName);
+  }
+
+  static async selectSourceAttachClonedDisk(PVCName: string, PVCNamespace: string) {
+    await selectOptionByText(view.diskSource, DISK_SOURCE.AttachClonedDisk);
```

Comment:
can't we always select the source inside the create/edit? It seems that every option is selecting it regardless.

Reply:
also, is there any benefit of using functional approach here? Wouln't it be more straightforward to just send the data? Could be inside an inner object like `sourceConfig: {}` ?

Reply:
I discussed possible solutions with Rasto, I wanted to avoid complicated logic in the `create` method. This seems like easier solution to maintain IMO, because we only need one mock (or one method generating disk object) for each disk type, so it doesn't lead to code duplicity.
On the other hand, I agree that having the configuration in the disk object as data is probably more readable. 

> can't we always select the source inside the create/edit? It seems that every option is selecting it regardless.
I'm not sure what you mean, not the way it's implemented now.

Reply:
ok, I guess both are valid approaches. I am a bit worried about this becoming a convention - passing unknown methods here and there where config could be used.

> I'm not sure what you mean, not the way it's implemented now.

I just meant that every option calls.
```tsx
await selectOptionByText(view.diskSource, DISK_SOURCE.Url);
```

 It would not have to be duplicated with the other approach.